# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/tests/crashtests/1704321-1.html
# Commit: 8f77fe5d56b2
# Full Hash: 8f77fe5d56b270ccadf6c361e40412a12e56d4b9
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2021-04-13 03:56:47
# Description:
#   Bug 1704321 - Handle non-invertible clip transforms for image clip masks. r=gfx-reviewers,bradwerth
#   
#   Not all transforms we encounter for image clip masks are invertible. When the unmap of the
#   world rect back to the local rect fails, we can safely fall back to using a scissor rect.
#   Skipping the scissor rect is an optimization and is not needed for correctness.
# ==============================================================================

diff -r c82840a55458 -r 8f77fe5d56b2 gfx/tests/crashtests/1704321-1.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/tests/crashtests/1704321-1.html	Mon Apr 12 18:58:20 2021 +0000
@@ -0,0 +1,13 @@
+<style>
+:root {
+  perspective: 1px;
+}
+#a {
+  -webkit-mask-image: url(data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs=);
+}
+:not(defs) {
+  transform: rotatex(1deg);
+}
+</style>
+<a style="-webkit-mask-image: url(#x)">
+<ul id="a">x</ul>